sefaz/accesskey/validation.go

```diff
@@ -5,14 +5,54 @@ import (
 	"github.com/arquivei/foundationkit/sefaz/stakeholder"
 )
 
+// Check checks if an access key is valid. It returns an error with an specific code based on
+// the validation problem
+func Check(accessKey AccessKey) error {
+	const op errors.Op = "accesskey.Check"
+
+	if accessKey == "" {
+		return errors.E(op, ErrEmptyAccessKey, ErrCodeEmptyAccessKey)
+	}
+
+	if len(accessKey) != 44 {
+		return errors.E(op, ErrInvalidLength, ErrCodeInvalidLength)
+	}
+
+	if !isDigitOnly(accessKey) {
+		return errors.E(op, ErrInvalidCharacter, ErrCodeInvalidCharacter)
+	}
+
+	if !isValidUF(accessKey[0:2].String()) {
+		return errors.E(op, ErrInvalidUF, ErrCodeInvalidUF)
+	}
+
+	if !isValidMonth(accessKey[4:6].String()) {
+		return errors.E(op, ErrInvalidMonth, ErrCodeInvalidMonth)
+	}
+
+	if !isValidCPFCNPJ(accessKey[6:20].String()) {
+		return errors.E(op, ErrInvalidCPFCNPJ, ErrCodeInvalidCPFCNPJ)
+	}
+
+	if !isValidModel(accessKey[20:22].String()) {
+		return errors.E(op, ErrInvalidModel, ErrCodeInvalidModel)
+	}
+
+	if !isValidationDigitCorrect(accessKey.String()) {
+		return errors.E(op, ErrInvalidDigit, ErrCodeInvalidDigit)
+	}
+
+	return nil
+}
+
 // validate execute all sub-routines necessary to perform a full accesskey validation
 func validate(accessKey AccessKey) error {
 	const op errors.Op = "validate"
 	if accessKey == "" {
 		return errors.E(op, ErrEmptyAccessKey, ErrCodeEmptyAccessKey)
 	}
 	if len(accessKey) != 44 {
-		return errors.E(op, ErrInvalidLenght, ErrCodeInvalidAccessKey)
+		return errors.E(op, ErrInvalidLength, ErrCodeInvalidAccessKey)
```

Comment:
I think here you should use `ErrCodeInvalidLength`

Reply:
The point here was to keep backward compatibility with the previous version. If we simply remove this code, that would be a breaking change once this is an Exported symbol.

Another approach would be deprecating this generic error code, what do you think?

Reply:
This is to keep backward compatibility